frontend/src/concepts/pipelines/utils.ts

```diff
@@ -1,10 +1,27 @@
-import { deletePipelineCR, deleteSecret } from '~/api';
-import { ExternalDatabaseSecret } from '~/concepts/pipelines/content/configurePipelinesServer/const';
+import { deletePipelineCR, deleteSecret, getPipelinesCR } from '~/api';
+import {
+  DSPA_SECRET_NAME,
+  ExternalDatabaseSecret,
+} from '~/concepts/pipelines/content/configurePipelinesServer/const';
 import { ELYRA_SECRET_NAME } from '~/concepts/pipelines/elyra/const';
 
-export const deleteServer = (namespace: string): Promise<void> =>
-  Promise.allSettled([
+export const deleteServer = async (namespace: string): Promise<void> => {
+  const dspa = await getPipelinesCR(namespace);
+  const dspaSecretName =
+    dspa.spec.objectStorage.externalStorage?.s3CredentialsSecret.secretName ?? '';
+  return Promise.allSettled([
     deleteSecret(namespace, ExternalDatabaseSecret.NAME),
     deleteSecret(namespace, ELYRA_SECRET_NAME),
+    deleteSecret(namespace, DSPA_SECRET_NAME),
+    isGeneratedDSPAExternalStorageSecret(dspaSecretName) && deleteSecret(namespace, dspaSecretName),
     deletePipelineCR(namespace),
   ]).then(() => undefined);
```

Comment:
If these don't exist, do we not error out? Should we mute the 404 errors for sanity?

Reply:
It's not error out at all, I guess it's because we are using `Promise.allSettled` instead of `Promise.all` so we don't need to catch it or verify the output.